src/node.rs

```diff
@@ -1224,42 +1252,112 @@ impl<S: Send + Sync + 'static + ForkSource + std::fmt::Debug> EthNamespaceT for
         let inner = Arc::clone(&self.inner);
 
         Box::pin(async move {
-            let reader = match inner.read() {
-                Ok(r) => r,
-                Err(_) => return Err(into_jsrpc_error(Web3Error::InternalError)),
+            let maybe_block = {
+                let reader = match inner.read() {
+                    Ok(r) => r,
+                    Err(_) => return Err(into_jsrpc_error(Web3Error::InternalError)),
+                };
+                match block_number {
+                    zksync_types::api::BlockNumber::Latest
+                    | zksync_types::api::BlockNumber::Pending
+                    | zksync_types::api::BlockNumber::Finalized
+                    | zksync_types::api::BlockNumber::Committed => reader
+                        .block_hashes
+                        .get(&reader.current_miniblock.saturating_sub(1))
+                        .and_then(|hash| reader.blocks.get(hash))
```

Comment:
Since we're now attempting to return directly from `reader.blocks`, we need to properly initialize that HashMap and  also the `block_hashes` HashMap. Else it will be empty for the first block and `BlockNumber::Latest` will never return anything.

I'm guessing that's why the e2e tests are failing